bee-protocol/src/worker/solidifier/milestone.rs

```diff
@@ -84,26 +85,28 @@ impl MilestoneSolidifierWorker {
 
         // TODO this may request unpublished milestones
         for index in solid_milestone_index..solid_milestone_index + MILESTONE_REQUEST_RANGE as u32 {
-            if !tangle().contains_milestone(index.into()) {
+            let index = index.into();
+            if !tangle().contains_milestone(index) {
                 Protocol::request_milestone(index, None);
             }
         }
     }
 
     async fn solidify_milestone(&self) {
-        let target_index = *tangle().get_solid_milestone_index() + 1;
+        let target_index = tangle().get_solid_milestone_index() + 1.into();
 
         if let Some(target_hash) = tangle().get_milestone_hash(target_index.into()) {
-            if tangle().is_solid_transaction(&target_hash) {
-                // TODO set confirmation index + trigger ledger
-                tangle().update_solid_milestone_index(target_index.into());
-                Protocol::broadcast_heartbeat(
-                    *tangle().get_solid_milestone_index(),
-                    *tangle().get_snapshot_milestone_index(),
-                )
-                .await;
-            } else {
-                Protocol::trigger_transaction_solidification(target_hash, target_index).await
+            match tangle().is_solid_transaction(&target_hash) {
```

Comment:
Any reason you prefer to match on a bool ? Clippy doesn't like it (and me neither!) :)

Reply:
Oh yea, that's stupid. That is a relic as this formerly returned an Option.